enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/ResetButton.tsx

```diff
@@ -1,20 +1,30 @@
-import { useMemo } from "react";
-
 import { useInteractiveQuestionContext } from "embedding-sdk/components/private/InteractiveQuestion/context";
 import { ResetButton } from "embedding-sdk/components/private/ResetButton";
+import { isSavedQuestionChanged } from "metabase/query_builder/utils/question";
+import * as Lib from "metabase-lib";
+
+export const QuestionResetButton = ({
+  onClick,
+}: {
+  onClick?: () => void;
+} = {}) => {
+  const { question, originalQuestion, onReset } =
+    useInteractiveQuestionContext();
 
-export const QuestionResetButton = () => {
-  const { question, onReset } = useInteractiveQuestionContext();
+  const handleReset = () => {
+    onReset?.();
```

Comment:
what do you mean here? like that we remove it from the provider entirely? or the prop?

Reply:
I suggest we change the mentioned line to
```ts
onReset: loadQuestion
```
And remove `onReset` from `InteractiveQuestionProvider` since we never pass it.

Reply:
ah ok. doing a small type-splitting right now so will do that as well

Reply:
and I just noticed how we use `onReset?.()`. I mean it's technically correct if you call this component outside the context provider then `onReset` would be undefined, but I feel like we should call it `onReset()` since we always set it in the context.

What do you think?